code/app/src/main/java/com/adobe/marketing/optimizeapp/SettingsScreen.kt

```diff
@@ -54,27 +62,147 @@ fun SettingsView(viewModel: MainViewModel) {
-            SettingsTextField(value = viewModel.textTargetPurchaseId, placeholder = "Enter Purchased Product Ids (comma-separated)") { viewModel.textTargetPurchaseId = it }
-            SettingsLabel(text = "Target Parameters - Product", align = TextAlign.Center, textStyle = MaterialTheme.typography.subtitle2)
-            SettingsTextField(value = viewModel.textTargetProductId, placeholder = "Enter Product Id") { viewModel.textTargetProductId = it }
-            SettingsTextField(value = viewModel.textTargetProductCategoryId, placeholder = "Enter Product Category id") { viewModel.textTargetProductCategoryId = it }
+            SettingsLabel(text = "Proposition Type", align = TextAlign.Start, textStyle = MaterialTheme.typography.subtitle1)
+            ExposedDropdownMenuBox(
+                expanded = expanded,
+                onExpandedChange = {
+                    expanded = !expanded
+                },
+                modifier = Modifier.absolutePadding(left = 20.dp, right = 20.dp, top = 10.dp).fillMaxWidth()
+            ) {
+                TextField(
+                    readOnly = true,
+                    placeholder = {
+                        Text(
+                            text = "Select Type",
+                        )
+                    },
+                    value =  viewModel.propositionType,
+                    onValueChange = { },
+                    trailingIcon = {
+                        ExposedDropdownMenuDefaults.TrailingIcon(
+                            expanded = expanded
+                        )
+                    },
+                    colors = TextFieldDefaults.textFieldColors(backgroundColor = Color.White, focusedIndicatorColor = Color.Transparent, unfocusedIndicatorColor = Color.Transparent),
+                    shape = RoundedCornerShape(size = 15.dp),
+                    modifier = Modifier.fillMaxWidth()
+                )
+                ExposedDropdownMenu(
+                    expanded = expanded,
+                    onDismissRequest = {
+                        expanded = false
+                    },
+                ) {
+                    propositionType.forEach { selectionOption ->
+                        DropdownMenuItem(onClick = {
+                                if (selectionOption != viewModel.propositionType) {
+                                    viewModel.clearCachedPropositions()
+                                }
+                                viewModel.propositionType = selectionOption
+                                expanded = false
+                            }
+                        ) {
+                            Text(text = selectionOption,
+                            )
+                        }
+                    }
+                }
+            }
+
+            if (viewModel.propositionType == "Custom Content") {
```

Comment:
nit: This is renamed `Custom Content` -> `Code-based Experiences`. Currently, you're also including Target in `ODE`. Can we rather just keep the types as "Surfaces" and "Decision Scopes".